server/ladder_service.py

```diff
@@ -374,7 +377,12 @@ async def start_game(
             game.init_mode = InitMode.AUTO_LOBBY
             game.map_file_path = map_path
 
-            for i, player in enumerate(alternate(team1, team2)):
+            # Shuffle the teams such that direct opponents remain the same
+            shuffled_teams = zip(*random.sample(
+                list(zip(team1, team2)),
+                len(team1)
+            ))
+            for i, player in enumerate(alternate(*shuffled_teams)):
```

Comment:
I guess this makes sense if the `alternate` method already exists for other reasons, but to me it looks like you have the result of `random.sample`, which is a list of tuples and you flatten it as `alternate(*zip(*random.sample(...)))` instead of e.g. `[player for player_pair in random.sample(...) for player in player_pair]`. I find the latter easier to reason about, but that might of course be a matter of taste :)

Reply:
Yea I think I like that better too. `alternate` made more sense before because the teams were in two separate lists. But since the best way I could think of for shuffling both lists in the same way was zipping them together, we have the result as a list of tuples anyways so it doesn't really make sense anymore.